ansible/roles/set-repositories/tasks/satellite-repos.yml

```diff
@@ -56,6 +56,13 @@
   copy:
     dest: /etc/rhsm/facts/katello.facts
     content: {"network.fqdn": "{{ inventory_hostname }}-{{ ansible_date_time.iso8601_basic | lower }}"}
+  when: cloud_provider == 'ec2'
+
+- name: Randomize hostname for Satellite
+  copy:
+    dest: /etc/rhsm/facts/katello.facts
+    content: {"network.fqdn": "{{ inventory_hostname }}.{{guid}}.internal-{{ ansible_date_time.iso8601_basic | lower }}"}
+  when: cloud_provider == 'osp'  
```

Comment:
For readability, i wouldn't use the inline dictionary type, but expand it.
And you can also do a single "randomize hostname for satellite" task with only the value changing depending on the cloud provider:

```yaml
- name: Randomize hostname for Satellite
  copy:
    dest: /etc/rhsm/facts/katello.facts
    content:
      network.fqdn: >-
        {% if cloud_provider == 'ec2' %}
        {{ inventory_hostname }}-{{ ansible_date_time.iso8601_basic | lower }}
        {% else %}
        {{ inventory_hostname }}.{{guid}}.internal-{{ ansible_date_time.iso8601_basic | lower }}
        {% endif %}
```

Or maybe the test should actually be about guid being already in the inventory hostname ?

```yaml
- name: Randomize hostname for Satellite
  copy:
    dest: /etc/rhsm/facts/katello.facts
    content:
      network.fqdn: >-
        {% if guid in inventory_hostname %}
        {{ inventory_hostname }}-{{ ansible_date_time.iso8601_basic | lower }}
        {% else %}
        {{ inventory_hostname }}.{{guid}}.internal-{{ ansible_date_time.iso8601_basic | lower }}
        {% endif %}
```

Reply:
@fridim , I have run several time test with three-tier-app config on osp and found node's hostname is not inclusive of guid.